### ISSUE TITLE
Reclaim Protocol - Bringing Web2 Identity to Web

### ISSUE THREAD

**Issue by roohchillsharma** [scrtlabs/Grants#122]
### **Project Description**

Reclaim Protocol makes https traffic verifiable using Zero-Knowledge Proofs, enabling users to generate verifiable credentials from any of their online user profiles. This unlocks unlimited possibilities as no APIs are required to extract the data or no changes to be made to the websites, while guaranteeing data integrity. 

### **Problem/Solution**

Currently, there is no way to bridge private user data from Web2 to Web 3. Oracles such as chainlink are effective at bridging publicly available data. However, private data that often resides behind a user login is inaccessible on-chain. e.g. number of Uber rides a user has taken is accessible only after the user has successfully logged in. This lack of a seamless bridge for private data between Web2 and Web3 creates a disconnect between user identities on different centralized platforms and restricts their utility on-chain across Dapps.

Because of Reclaim, Web2 user data which was elusive to Web3 till now will be available across dapps on Secret. This opens up opportunities for a new wave of applications in privacy, sybil resistance, proof of personhood, KYC and many more. Dapp developers can leverage Reclaim protocol to build multiple use cases related to **KYC, sybil resistance, proof of personhood, etc** on top of Secret using Reclaim. 
****

### **Detailed Project Overview**

Reclaim empowers users to generate Zero-Knowledge Proofs for any online user profile. To generate a claim, users first need to log into the relevant website. This login process, involving an HTTPS request and its subsequent response, is channeled through an HTTPS Proxy Server known as an 'attestor'. This attestor oversees the encrypted data exchange between the user and the website. Subsequently, users provide keys that disclose non-sensitive parts of the request to the attestor. With this, the attestor can view the request in its entirety, barring confidential details like authentication data, and can confirm its legitimacy.

The website's encrypted response is then processed by a zk-circuit, which identifies a regex match within the encrypted data using a decryption key as a confidential input. The attestor further validates that the zk-circuit's public input was indeed the encrypted data source from the website. With these attestations on both the request and the encrypted response, coupled with the zk-proof, any third-party application, whether on-chain or off-chain, can verify the existence of data that exists on the user’s profile.

<img width="1050" alt="Reclaim Sequence Diagram" src="https://github.com/scrtlabs/Grants/assets/83450876/ae99c895-2dca-4a84-ae15-2784cca31712">

<img width="715" alt="Zk Circuit" src="https://github.com/scrtlabs/Grants/assets/83450876/254287ff-5691-438a-960e-e476e57acc44">

User credentials in Reclaim Protocol are generated and stored completely on the client side. **Using Reclaim, users can generate Proofs(Groth16) in less than 10 secs even on a 2015 Android Device!** 

### **Project Links**

Product Demo - [Link](https://www.loom.com/share/b88ab8bb90ff498c8cbd486bdb4a42a4)

Reclaim Protocol - https://github.com/reclaimprotocol

Questbook - https://github.com/questbook

SDK - https://docs.reclaimprotocol.org/

Whitepaper - https://drive.google.com/file/d/1wmfdtIGPaN9uJBI1DHqN903tP9c_aTG2/view

Website - https://www.reclaimprotocol.org/

Explainer Blogposts - https://blog.reclaimprotocol.org/

### **Go to Market Plan**

1. **Identifying Unique Use Cases Enabled by Reclaim**
Upon integrating with the Secret Network, we will collaborate closely with the Secret Network team, foundation, and community to identify specific use cases, Dapps (apart from [these](https://www.notion.so/Secret-Reclaim-0284dcd205864ca2b1ea058bc2d0125e?pvs=21) that are uniquely possible due to Reclaim to accelerate Secret Network’s growth. 

1. **Educational Initiatives**
We will host webinars, workshops, and online documentation to educate builders about Reclaim and our SDK integration steps, use cases that Reclaim unlocks, and offer any technical support. We will start with our already established community of high quality developers on [Questbook.app](https://www.questbook.app/)

2. **Active  Participation in Developer Community Calls and Events**
To further drive the awareness and development of novel use cases with Reclaim on top of Secret Network, we will actively organise and participate in all developer community calls. Our involvement will focus on facilitating the development of identified use cases, thereby boosting both developer engagement and user adoption on the Secret Network.

3. **Marketing and Promotion Strategy**:
    - **Content Marketing**: We will create and distribute informative content such as blog posts, whitepapers, case studies, and videos that demonstrate Reclaim's capabilities and benefits
    - **Social Media Marketing**: We will actively engage with the builder community on platforms such as Twitter, LinkedIn, Reddit, and Telegram to promote Dapps that have grown because of Reclaim.
    

### **Grant Proposal - Integrate Reclaim with Secret**

As part of this proposal, we will implement the following: 

- Research Secret Network in detail, its design, architecture and share implementation specification documents with the Secret Network team for feedback and comments
- Integrate Reclaim Protocol with Secret
- Create documentation to help builders in the Secret ecosystem on using Reclaim SDK in their dapps
- A blog post and podcast to detail Reclaim protocol and its benefits to Secret

### **Why Secret Network?**

We are a small team with finite resources, and want to collaborate with ecosystems that demonstrate a genuine commitment to enable use cases that are likely to scale to a billion users. Specifically, we want to build on ecosystems that exhibit a strong desire to leverage Web2 sources to provide seamless user onboarding. To our knowledge, there aren’t that many ecosystems such as Secret Network that have put a strong emphasis on user privacy, streamlining user onboarding and experience 

### **Value Capture to Secret Network**

Reclaim offers several advantages to the Secret Network ecosystem. Some of the benefits are listed as follows: 

1. **Enhanced Interoperability** 
- By allowing Dapps on Secret Network to access data from Web2 platforms, we can significantly expand the range of applications and use cases that can be developed on Secret Network, especially those focussed around KYC, sybil resistance, proof of personhood, thus driving significant ecosystem growth. Additionally, Reclaim will empower existing Dapp developers of Dapps such as [Bushi, BlackJack](https://scrt.network/ecosystem/dapps) and new Dapp developers to expand their use cases and deliver an improved user experience to their end-users.

1. **Network Growth and Adoption** 
- As Reclaim brings its unique capabilities to Secret Network, it will drive the growth of the network significantly. More users and use cases will translate to increased transactions and engagement, enhancing the network's value and utility. Moreover, Reclaim’s ability to facilitate access to off-chain data with user consent and cryptographic assurance of data integrity aligns with Secret Network’s commitment to user-centricity and security.
1. **Enabling Novel Dapps and Use-Cases:**
- By facilitating access to off-chain user data, Reclaim will enable the development of [new, innovative use cases](https://blog.reclaimprotocol.org/posts/usecases/) on Secret Network, such as Gaming IDs, P2P onramp/offramp, Gossip marketplaces, or social Dapps that utilize user data from Web2 platforms.

1. **Enhanced Developer Mindshare:** 
- By facilitating new and unique use cases that leverage off-chain data, Reclaim protocol will attract a broader developer community to Secret Network, enriching its ecosystem and increasing its competitive edge

### **Grant Request Overview**

$40,000

**Funding Milestones**

**Milestone - 1**
Project Kickoff & grant approval - $5,000 

**Milestone - 2** 

Reclaim Secret Integration - $20,000

- Successfully deploying of Reclaim Contracts
- Effectively deploying Semaphore Contracts
- Optimizing zkProof verification for Secret’s constraints (e.g. speed/gas constraints)
- Integrating and Testing proof submission through SDKs, ensuring reliable and consistent interactions with Secret

**Milestone - 3**

Comprehensive documentation, blog posts, and community calls within the Secret ecosystem - $5,000

**Milestone - 4**
Launch of 2 Dapps on production within the Secret ecosystem that integrate Reclaim protocol - $10,000

**Detailed Breakdown**

To accomplish the milestones outlined above, the entire project will require 1 Project Manager, 3 Developers, 1 Dev Relations Lead, and 1 Integration Support Developer.

Integration - 3 Developers * 80 hrs * $90 = ~$20,000

Project Manager - 80 hrs * $120 ~ $10,000

Dev Relations & Post Integration Support - 2 * 50 hrs * $100 = $10,000

We will offer the Secret Nework’s ecosystem ongoing development support for three months post-integration led by Integration Support Developer and provided by Reclaim Protocol. Additionally, after the integration is successfully complete, we'll require assistance from the Secret Network team for joint marketing efforts to increase Reclaim’s visibility among developers and encourage developers to build novel use cases on top of Secret Network using Reclaim. Some of the use cases can be from the following - https://blog.reclaimprotocol.org/posts/usecases/

### **Current Focus Areas**

1. With Reclaim protocol now live on production, we are working towards helping reputed ecosystems such as Secret that are exploring identity solutions build stronger infrastructure and subsequently, fast forward their growth. 

2. Working with experienced and high-quality Dapp developers to integrate or build novel use cases using Reclaim Protocol. Some of the examples include **[decentralized P2P exchange](https://www.p2px.finance/),** which has already done $150,000 in volumes in just three weeks,  [Whistleblower](https://whistleblower-frontend-acez.onrender.com/), and [Cupid Coin](https://www.reclaimprotocol.org/#:~:text=Cupid%20coin%20takes%20a%20whole%20new%20take%20on%20a%20community%20owned%20Dating%20App.%20To%20bootstrap%20the%20app%2C%20it%20allows%20only%20high%20quality%20users.%20Only%20users%20with%20greater%20than%2025%20matches%20on%20Tinder/Bumble%20are%20allowed%20to%20join.%20Before%20this%2C%20there%20was%20no%20way%20for%20them%20to%20get%20the%20users%27%20match%20count%20on%20other%20apps) 

3. **Perfecting Developer Experience** - Providing a seamless Reclaim integration experience to external developers and users using Reclaim by streamlining various touchpoints such as developer documentation, SDKs, developer journey, user journey etc. 

### **Development Roadmap**

Our long term vision is to **unlock a new design space empowering builders** to build novel solutions that leverage the identity and credentials of users which they have accrued over time all over the internet. We are also keen to engage closely with the ecosystems/partners who share our vision and can help us realise it. 

To accomplish this, our immediate goal is to provide a seamless Reclaim integration experience to all developers and users using Reclaim by streamlining its various touchpoints such as developer documentation, Reclaim SDKs, developer journey, user journey  etc. 

### **Team Details**

Reclaim Protocol is built by the team at CreatorOS Inc. We are a 20+ member engineering and web3 product development & research team including ZKP researchers and with previous affiliations to Stanford, Microsoft, Meta and Google . We have also built - [Questbook.app](http://Questbook.app), an industry leading on-chain grants management tool that is used by some of the major L1/L2s including Polygon, Solana, Compound, Arbitrum, Ton, among others.  CreatorOS is a YC W21 company.

- Madhavan Malolan : CEO
    - Building in crypto since 2016.
    - Among first 5 contributors to Plasma (ethereum scaling solution) specifications.
    - Open source contributor.
    - ex-Microsoft, Computer Science IIIT-H.
    - [LinkedIn](https://www.linkedin.com/in/madhavanmalolan/) [Github](https://github.com/madhavanmalolan)

- Abhilash Inumella : Co-founder, Leads Product
    - Building in Crypto since 2019.
    - ex-CEO of Samosa Labs (10M users, funded by Sequoia, Xiaomi).
    - Ex-Google, Ex-Facebook, Computer Science IIIT-H.
    - [LinkedIn](https://www.linkedin.com/in/abhilash-inumella-89a92421/).
    
- Max Allman, Mechanism Design Researcher
    - PhD from Stanform in Mechanism Design and Game Theory
    - Co- author of the [Reclaim Whitepaper](https://drive.google.com/file/d/1wmfdtIGPaN9uJBI1DHqN903tP9c_aTG2/view)

- [Kirill Kutsenok](https://www.linkedin.com/in/kirill-kutsenok/), Cryptography & Security Researcher
- [Adhiraj](https://www.linkedin.com/in/adiwajshing/) Singh: Lead Developer
- [Sweta](https://www.linkedin.com/in/sweta-shaw/) Shaw: Developer Relations
- Aleksai Ermishkin: Lead Blockchain Developer
